File: src/app/models/DecisionNode.ts
export interface DecisionNode {
  question: Question;
  answers: Answer[];
}

export interface Question {
  id: string;
  question: string;
  answers: Answer[];
  multipleChoices: boolean;
  category: string;
  description: string;
}

export interface Answer {
  text: string;
  labels: string[];
  next: string;
  description: string;
}

File: src/main/java/uk/ac/ebi/fairwizard/model/DecisionNode.java
package uk.ac.ebi.fairwizard.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.persistence.Id;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document
public class DecisionNode {
  @Id
  private String id;
  private String question;
  private String category;
  private String description;
  private boolean multipleChoices;
  private List<Answer> answers;
}
